### PR TITLE
Add ISBN search to WHO translator

### LINKED ISSUE #2617: WHO ISBN search translator
https://forums.zotero.org/discussion/91092/whos-iris-pdf-metadata-automatic-citation-download-anyone-would-want-to-develop

I think we can just reject detection when the ISBN doesn't start with "978924". (This seems like something we should do for more things?)

I think we can just put this in the existing WHO translator and make it translatorType 12.

### PR DESCRIPTION
**Pull request by AbeJellinek**
Scaffold will delete search tests, so none added.

Fixes #2617.

### CODE REVIEW

File: WHO.js [hunk 1]
```diff
@@ -154,6 +149,36 @@ function scrape(doc, url) {
 	});
 }
 
+function detectSearch(item) {
+	return item.ISBN && ZU.cleanISBN(item.ISBN).startsWith('978924');
```
Comment (dstillman) [inline WHO.js:155]: `cleanISBN` can return false, so this would probably throw for an invalid ISBN.

File: WHO.js [hunk 2]
```diff
@@ -154,6 +149,36 @@ function scrape(doc, url) {
...
+		translator.setHandler("itemDone", function (obj, item) {
+			item.libraryCatalog = 'WHO IRIS';
+			item.archive = '';
+			
+			if (item.url) {
+				ZU.processDocuments(item.url, function (recordDoc) {
+					item.attachments.push({
+						title: 'Full Text PDF',
+						mimeType: 'application/pdf',
+						url: attr(recordDoc, 'meta[name="citation_pdf_url"]', 'content')
```
Comment (dstillman) [inline WHO.js:174]: Probably should test this before pushing in case it's missing.

File: WHO.js [hunk 3]
```diff
@@ -154,6 +149,36 @@ function scrape(doc, url) {
 	});
 }
 
+function detectSearch(item) {
+	return item.ISBN && ZU.cleanISBN(item.ISBN).startsWith('978924');
+}
+
+function doSearch(item) {
+	let ISBN = ZU.cleanISBN(item.ISBN);
+	let url = `https://apps.who.int/iris/discover/export?format=refman&list=discover&rpp=10&etal=0&query=${ISBN}&group_by=none&page=1`;
```
Comment (dstillman) [inline WHO.js:160]: Is the RIS quality comparable to the EM we get normally? If it's any lossier, it seems like we should just use the standard scraping routine.

File: WHO.js [hunk 4]
```diff
@@ -148,12 +145,42 @@ function scrape(doc, url) {
...
+	ZU.doGet(url, function (risText) {
```
Comment (dstillman) [inline WHO.js:161]: At least from curl, it looks like this returns an empty 200 response in the case of an unknown ISBN. Not sure what the RIS translator does on an empty string, but maybe cleaner to test for an empty `risText` here and bail.

File: WHO.js (hunk 3 above)
Comment (AbeJellinek) [reply WHO.js:160]: It's more or less equivalent. EM is missing books' editions, the RIS has `numPages` in the wrong field (but I'll fix that). I think RIS makes more sense for a search translator.

File: WHO.js (hunk 4 above)
Comment (AbeJellinek) [reply WHO.js:161]: The RIS translator just returns no items when it gets an empty string, so I don't think bailing will do anything but save us a little time loading the translator. But there's no reason not to, so I'll add it.

**Comment by dstillman** [comment]
👍

**Pull request merged**